checkov/common/checks/base_check.py

```diff
@@ -113,3 +114,6 @@ def get_evaluated_keys(self) -> List[str]:
 
     def get_output_id(self, use_bc_ids: bool) -> str:
         return self.bc_id if self.bc_id and use_bc_ids else self.id
+
+    def add_detail(self, msg: str):
+        self.details.append(msg)
```

Comment:
you can remove the method, because the attribute `details` is not defined as private and therefore can and should be accessed directly, like `self.details.append(msg)` inside the actual check.